### PR TITLE
Support uploading attachments

### PR DESCRIPTION
**Pull request by Trolldemorted**
Support sending attachments by
- clicking the attach icon
- dragndropping a file into the conversation
- pressing ctrl+v if the buffer is filled with a screenshot

### CODE REVIEW

File: Signal-Windows.Lib/OutgoingMessages.cs [hunk 1]
```diff
@@ -68,11 +70,29 @@ public void SendMessage(SignalServiceSyncMessage message)
                 try
                 {
                     outgoingSignalMessage = Handle.OutgoingQueue.Take(Token);
+                    List<SignalServiceAttachment> outgoingAttachmentsList = null;
+                    if (outgoingSignalMessage.Attachments != null && outgoingSignalMessage.Attachments.Count > 0)
+                    {
+                        outgoingAttachmentsList = new List<SignalServiceAttachment>();
+                        foreach (var attachment in outgoingSignalMessage.Attachments)
+                        {
+                            var file = await ApplicationData.Current.LocalCacheFolder.GetFileAsync(@"Attachments\" + attachment.Id + ".plain");
```
Comment (golf1052) [inline Signal-Windows.Lib/OutgoingMessages.cs]: If the `Attachments` doesn't exist this throws an exception (FileNotFound). Need to check that the Attachments folder exists and if it doesn't create it first.

File: Signal-Windows.Lib/SignalLibHandle.cs [hunk 2]
```diff
@@ -512,7 +536,15 @@ public void StartAttachmentDownload(SignalAttachment sa)
             SignalDBContext.SaveMessageLocked(message);
             conversation.LastMessage = message;
             conversation.LastActiveTimestamp = message.ComposedTimestamp;
-            //StartAttachmentDownloads(message);
+            if (attachmentStorageFile != null)
+            {
+                StorageFolder plaintextFile = await ApplicationData.Current.LocalCacheFolder.GetFolderAsync(@"Attachments\");
```
Comment (golf1052) [inline Signal-Windows.Lib/SignalLibHandle.cs]: Same comment as above. Need to make sure `Attachments` folder exists.

Comment (Trolldemorted) [reply Signal-Windows.Lib/SignalLibHandle.cs]: yup, we should indeed use `CreateFolderAsync` with `OpenIfExists` here

File: Signal-Windows.Lib/OutgoingMessages.cs (hunk 1 above)
Comment (Trolldemorted) [reply Signal-Windows.Lib/OutgoingMessages.cs]: Nice catch! We should also handle if the entire attachment plaintext is missing, though

**Pull request merged**